### PR TITLE
feat: add support for displaying type information of interfaces

### PR DESCRIPTION
**Pull request by ftr93**

### CODE REVIEW

File: packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/ast-utils.ts [hunk 1]
```diff
@@ -27,12 +39,25 @@ export function extractComponentOrDirectiveAnnotatedClasses(
 					classDeclaration.getDecorator('Directive') ||
 					classDeclaration.getDecorator('Pipe') ||
 					classDeclaration.getDecorator('Injectable')
-				),
+				) || hasJsDocsTag(classDeclaration.getJsDocs(), 'include-docs'),
```
Comment (Etienne-Buschong) [inline packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/ast-utils.ts:42]: Does this make sense? Are there non Component/Directive/Pipe/Injectable classes that we support?

On the other hand, it does not really hurt to have it here 🤷

File: packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/ast-utils.ts [hunk 2]
```diff
@@ -27,12 +39,25 @@ export function extractComponentOrDirectiveAnnotatedClasses(
...
+		)
+		.filter(
+			(classDeclaration: ClassDeclaration) =>
+				!hasJsDocsTag(classDeclaration.getJsDocs(), 'exclude-docs'),
 		)
 		.filter((classDeclaration: ClassDeclaration) => !classDeclaration.isAbstract())
```
Comment (Etienne-Buschong) [inline packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/ast-utils.ts:48]: Can be done in a single filter statement

File: packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/interface-type-extraction.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	propertiesToExclude: RegExp | undefined,
+): Map<string, Entity> {
+	const properties = interfaceDeclaration.getProperties();
+	const methods = interfaceDeclaration.getMethods();
+	const entities = new Map<string, Entity>();
+
+	for (const signature of [...properties, ...methods]) {
+		if (isBuiltinAngularMethod(signature.getName())) {
+			continue;
+		}
```
Comment (Etienne-Buschong) [inline packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/interface-type-extraction.ts:22]: Not really relevant for interfaces. We should probably remove this?

File: packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/ast-utils.ts (hunk 1 above)
Comment (ftr93) [reply packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/type-extraction/ast-utils.ts:42]: I thought about adding ArgsTables for DataSources. And also we allow any interface with "@include-docs", so why not classes?

**Pull request closed without merging**